### PR TITLE
feat: clear tmp files

### PR DESCRIPTION
**Pull request by gabrielfs7**
https://oat-sa.atlassian.net/browse/ADF-237

### CODE REVIEW

File: model/MediaSource.php [hunk 1]
```diff
@@ -54,6 +54,9 @@ class MediaSource extends Configurable implements MediaManagement, ProcessedFile
     /** @var MediaBrowserPermissionsMapper */
     private $permissionsMapper;
 
+    /** @var string[] */
+    private $tmpFiles;
```
Comment (boajer) [inline model/MediaSource.php:58]:
```suggestion
    private $tmpFiles = [];
```

File: model/MediaSource.php [hunk 2]
```diff
@@ -428,4 +434,13 @@ private function getPermissionsMapper(): MediaBrowserPermissionsMapper
 
         return $this->permissionsMapper;
     }
+
+    public function __destruct()
+    {
+        foreach ($this->tmpFiles as $tmpFile) {
+            if (file_exists($tmpFile) && is_writable($tmpFile)) {
```
Comment (boajer) [inline model/MediaSource.php:441]:
```suggestion
            if (is_writable($tmpFile)) {
```
As https://www.php.net/manual/en/function.is-writable.php `Returns true if the filename exists and is writable`

**Review by chinnu-focaloid — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [X] Acceptance criteria are respected
- [X] Pull request title and description are meaningful

**Review by voknil — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [x] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [x] Pull request title and description are meaningful

**Pull request merged**